src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -259,417 +259,441 @@ AND NOT EXISTS (
+        Join<Pool, Product> product, SetJoin<Product, Product> providedProduct, Set<String> productIds) {
+
+        if ((productIds == null || productIds.isEmpty())) {
+            return Optional.empty();
         }
 
-        if (filters != null) {
-            // Pool ID filters
-            Collection<String> idFilters = filters.getIdFilters();
+        if (providedProduct == null) {
+            providedProduct = product.join(Product_.providedProducts, JoinType.LEFT);
+        }
 
-            if (idFilters != null && !idFilters.isEmpty()) {
-                criteria.add(CPRestrictions.in("Pool.id", idFilters));
-            }
+        List<Predicate> predicates = new ArrayList<>();
+        buildQueryArgumentInPredicate(product.get(Product_.id), productIds)
+            .ifPresent(predicates::add);
 
-            // Matches stuff
-            Collection<String> matchesFilters = filters.getMatchesFilters();
-            if (matchesFilters != null && !matchesFilters.isEmpty()) {
-                if (!joinedProvided) {
-                    // This was an inner join -- might end up being important later
-                    criteria.createAlias("Product.providedProducts", "Provided",
-                        JoinType.LEFT_OUTER_JOIN);
-                    joinedProvided = true;
-                }
+        buildQueryArgumentInPredicate(providedProduct.get(Product_.id), productIds)
+            .ifPresent(predicates::add);
 
-                criteria.createAlias("Provided.productContent", "PPC", JoinType.LEFT_OUTER_JOIN);
-                criteria.createAlias("PPC.content", "Content", JoinType.LEFT_OUTER_JOIN);
+        if (predicates.isEmpty()) {
+            return Optional.empty();
+        }
 
-                for (String matches : matchesFilters) {
-                    String sanitized = this.sanitizeMatchesFilter(matches);
+        return Optional.of(builder.or(predicates.toArray(new Predicate[0])));
+    }
 
-                    Disjunction matchesDisjunction = Restrictions.disjunction();
+    private Optional<List<Predicate>> getMatchesPredicates(CriteriaQuery<?> query, Root<Pool> root,
+        CriteriaBuilder builder, Join<Pool, Product> product, SetJoin<Product, Product> providedProduct,
+        Set<String> matches) {
 
-                    matchesDisjunction.add(CPRestrictions.ilike("Pool.contractNumber", sanitized, '!'))
-                        .add(CPRestrictions.ilike("Pool.orderNumber", sanitized, '!'))
-                        .add(CPRestrictions.ilike("Product.id", sanitized, '!'))
-                        .add(CPRestrictions.ilike("Product.name", sanitized, '!'))
-                        .add(CPRestrictions.ilike("Provided.id", sanitized, '!'))
-                        .add(CPRestrictions.ilike("Provided.name", sanitized, '!'))
-                        .add(CPRestrictions.ilike("Content.name", sanitized, '!'))
-                        .add(CPRestrictions.ilike("Content.label", sanitized, '!'))
```

Comment:
CPRestrictions was only used in PoolCurator. Now that its no longer used we can delete it along with CPRestrictionsTest.java

Reply:
Good catch, removed both CPRestrictions.java and CPRestrictionsTest.java